packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts

```diff
@@ -71,22 +71,25 @@ export function useUploadFiles({
       };
 
       if (file) {
-        resolveFile({ processFile, file, key }).then(({ key, file }) => {
-          onUploadStart?.({ key });
-          const uploadTask = uploadFile({
-            file,
-            key,
-            level: accessLevel,
-            progressCallback: onProgress,
-            errorCallback: onError,
-            completeCallback: onComplete,
-          });
+        resolveFile({ processFile, file, key }).then(
+          ({ key, file, ...rest }) => {
+            onUploadStart?.({ key });
+            const uploadTask = uploadFile({
+              file,
+              key,
+              level: accessLevel,
+              progressCallback: onProgress,
+              errorCallback: onError,
+              completeCallback: onComplete,
+              ...rest,
```

Comment:
I think previously we spread first, so we didn't accidentally override anything?

Reply:
good catch. fixed.